### ISSUE TITLE
DOC: one dimension can be -1 works also for numpy.ndarray.shape.

### ISSUE THREAD

**Issue by solarjoe** [numpy/numpy#9539]
The documentation for `np.reshape` states
> One shape dimension can be -1. In this case, the value is inferred from the length of the array and remaining dimensions.

The same is also true for changing the shape using `numpy.ndarray.shape`
```
import numpy as np
x = np.array([1, 2, 3, 4])
x.shape = (2, 2)
print(x)
x.shape = (-1, 4)
print(x)
```
But it is not listed [in its doc](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.shape.html).

Is this an "official" feature? If yes, I will adjust the doc.

**Comment by seberg** [comment]
That code abuses reshape internally, so yeah, you can say it is official since we should not change it, people may rely on it anyway.

The `note` about `reshape`... If you update the docs, please add `reshape` to the see also page. Also, please write that `reshape` should be always preferred, since setting the shape is not really a safe thing to do. So, at least in documentation, this feature should be deprecated IMO (or warned about).

There ** may be very** few exceptions where this makes sense (all hopefully only for arrays created just a bit earlier). Reshaping will (if this works) also create a view. Creating a view is safer and very fast. If you change an array in-place, you might screw up another part of a program which holds a reference to the same array and assumes that shapes cannot change.

So again, yes, you can document it. But also, please don't use it unless you have a very good excuse for yourself.

**Comment by solarjoe** [comment]
https://github.com/solarjoe/numpy/pull/1

**Comment by solarjoe** [comment]
https://github.com/numpy/numpy/pull/9569

**Comment by solarjoe** [comment]
https://github.com/numpy/numpy/pull/9571

**Comment by eric-wieser** [comment]
Fixed in #9810
